openerp/addons/test_new_api/models.py

```diff
@@ -143,12 +143,17 @@ class Discussion(models.Model):
     participants = fields.Many2many('res.users')
     messages = fields.One2many('test_new_api.message', 'discussion')
     message_changes = fields.Integer(string='Message changes')
-    important_messages = fields.One2many('test_new_api.message', 'discussion',
-                                         domain=[('important', '=', True)])
+    important_messages = fields.One2many(
+        'test_new_api.message', 'discussion',
+        domain=lambda self: self._domain_important())
     emails = fields.One2many('test_new_api.emailmessage', 'discussion')
     important_emails = fields.One2many('test_new_api.emailmessage', 'discussion',
                                        domain=[('important', '=', True)])
 
+    def _domain_important(self):
+        """Ensure computed O2M domains work as expected."""
+        return [("important", "=", True)]
+
```

Comment:
Can you please revert the changes on field `important_messages`, and create a new field for this case? Your modification actually discards the original test: invalidating the cache before checking its consistency!

I suggest another, equivalent, field like
```python
# ensure that computed o2m domains work as expected
very_important_messages = fields.One2many('test_new_api.message', 'discussion',
                                          domain=lambda self: self._domain_very_important())

def _domain_very_important(self):
    return [('important', '=', True)]
```